From fc5f86edda83e93912ff3403b833c2358d5d3ec3 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:37:08 -0400
Subject: [scsi] remove a useless get_put_device pair in scsi_next_command

Message-id: <1402670238-13916-141-git-send-email-snitzer@redhat.com>
Patchwork-id: 83877
O-Subject: [RHEL7.1 PATCH 140/150] scsi: remove a useless get_put_device pair in scsi_next_command
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

BZ: 1105204

Upstream commit 27e9e0f12acb5a76c72c3eed31ad3336e1970717
Author: Bart Van Assche <bvanassche@acm.org>
Date:   Thu Feb 20 14:20:57 2014 -0800

    [SCSI] remove a useless get/put_device pair in scsi_next_command

    Eliminate a get_device() / put_device() pair from scsi_next_command().
    Both are atomic operations hence removing these slightly improves
    performance.

    [hch: slight changes due to different context]
    Signed-off-by: Bart Van Assche <bvanassche@acm.org>
    Reviewed-by: Hannes Reinecke <hare@suse.de>
    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: James Bottomley <JBottomley@Parallels.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/scsi_lib.c b/drivers/scsi/scsi_lib.c
index a7d6a2d..09b97d8 100644
--- a/drivers/scsi/scsi_lib.c
+++ b/drivers/scsi/scsi_lib.c
@@ -549,14 +549,9 @@ void scsi_next_command(struct scsi_cmnd *cmd)
  struct scsi_device *sdev = cmd->device;
  struct request_queue *q = sdev->request_queue;
 
- /* need to hold a reference on the device before we let go of the cmd */
- get_device(&sdev->sdev_gendev);
-
  scsi_put_command(cmd);
- put_device(&sdev->sdev_gendev);
  scsi_run_queue(q);
 
- /* ok to remove device now */
  put_device(&sdev->sdev_gendev);
 }
 
-- 
1.7.1